src/Symfony/Component/Lock/Store/SemaphoreStore.php

```diff
@@ -63,7 +63,7 @@ private function lock(Key $key, bool $blocking)
             return;
         }
 
-        $keyId = crc32($key);
+        $keyId = unpack('i', md5($key, true))[1];
```

Comment:
What is the purpose of this change?

Don't we have the same code in Lock components?

Reply:
A crc32 is a 32bits number.
This new code uses a 64bits number on x64.
This reduces the risk of collision (that could cause a deadlock) by a huge amount.

Reply:
And this *is* the Lock component :)

Reply:
this could be mentioned in the commit message, to give context to the change in the git history